src/rpft/parsers/sheets.py

```diff
@@ -39,30 +44,34 @@ def get_sheets_by_name(self, name):
         return [sheet]
 
 
-class CSVSheetReader(AbstractSheetReader):
-    def __init__(self, filename):
-        self.name = filename
-        self.path = os.path.dirname(filename)
-        with open(filename, mode="r", encoding="utf-8") as table_data:
-            self.main_sheet = tablib.import_set(table_data, format="csv")
+class CSVSheetReader():
+    def __init__(self, path, main):
+        self.path = Path(path)
+        self.main = main
+        self._sheets_by_name = {f.stem: load_csv(f) for f in path.glob("*.csv")}
```

Comment:
Going along with this for now, but...

I think the abstract class is not giving us much value and it is dictating implementation details to classes that inherit from it. I'm also not on-board with returning several main sheets or several sheets by name, but I would have to know more about the parsers before I can suggest an alternative.

Reply:
I see the point, but I'd rather avoid the code duplication so that I can be confident that the Mock actually behaves like the actual classes.

If we were to only return a single sheet by name/as main, then the individual sheet readers and the composite can't be used interchangeably (also I'm not sure whether it's possible to have multiple sheets of the same name within an XLSX/Google sheet). I think this is nice to have (even if it could be worked around if we didn't have it).